docs/api/rest_api_usage.md

```diff
@@ -0,0 +1,697 @@
+
+### Request body
+
+While some short scalar parameters can be passed in the URIs or as GET parameters, some resources needs heavier structured payloads passed in the request body, in particular the ones to create (`POST`) or update (`PATCH`) items.
+In the [REST API reference](rest_api_reference/rest_api_reference.html), request payload examples are given when needed.
+
+One example is the [creation of an authentication session](rest_api_authentication.md#establishing-a-session).
+
+When creating a Content, the payload is particular if the ContentType has some [Image](field_types_reference/imagefield.md) or [BinaryFile](field_types_reference/binaryfilefield.md) fields as files need to be attached. See the example of a [script uploading images](#creating-content-with-binary-attachments) below.
+
+When searching for Contents (or Locations), the query grammar is also particular. See the [Search section](#search-view) below.
+
+#### Creating content with binary attachments
+
+The example below is a command-line script to upload images.
+
+This script will
+
+- receive an image path and optionally a name as command-line arguments,
+- use the [HTTP basic authentication](rest_api_authentication.md#http-basic-authentication) assuming it is enabled,
+- create a draft in the /Media/Images folder by `POST`ing data to [`/content/objects`](rest_api_reference/rest_api_reference.html#managing-content-create-content-item),
+- and, `PUBLISH` the draft through [`/content/objects/{contentId}/versions/{versionNo}`](rest_api_reference/rest_api_reference.html#managing-content-publish-a-content-version).
+
+=== "XML"
+
+    ``` php
+    [[= include_file('code_samples/api/rest_api/create_image.xml.php', 0, None, '    ') =]]
+    ```
+
+=== "JSON"
+
+    ``` php
+    [[= include_file('code_samples/api/rest_api/create_image.json.php', 0, None, '    ') =]]
+    ```
+
+#### Search (`/view`)
+
+The `/view` route allow to [search in the repository](../guide/search/search.md). It works close to its [PHP API counterpart](public_php_api_search.md).
+
+The model allows combining criteria using the logical operators `AND`, `OR` and `NOT`.
+
+Almost all [search criteria](../guide/search/criteria_reference/search_criteria_reference.md#search-criteria) are available on REST API. The suffix `Criterion` is added when used with REST API.
+
+Almost all [sort clauses](../guide/search/sort_clause_reference/sort_clause_reference.md#sort-clauses) are available too. No prefix ou suffix for them.
+
+The search request HTTP header to type its body is `Content-Type: application/vnd.ibexa.api.ViewInput+xml` or `+json`.
+The root node is `<ViewInput>` and it has two mandatory children: `<identifier>` and `<Query>`.
+
+`version=1.1` can be added to the `Content-Type` header to support the distinction between `ContentQuery` and `LocationQuery` instead of just `Query` which implicitly looked only for Contents.
```

Comment:
```suggestion
You can add `version=1.1` to the `Content-Type` header to support the distinction between `ContentQuery` and `LocationQuery` instead of just `Query` which implicitly looked only for Content items.
```

Reply:
```suggestion
You can add `version=1.1` to the `Content-Type` header to support the distinction between `ContentQuery` and `LocationQuery` instead of just `Query` which implicitly looks only for Content items.
```